### PR TITLE
TGA image origin and support decoding 16bit monochome images

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

This PR fixes two issues.
- Images with a bottom right or top right origin now decode correctly.
- Adds support for decoding 16bit gray images.

There are 7 failing tests, but it looks to me that the reference output from Magick.Net is not correct.
Here is an example for image `grayscale_a_UL.tga`, which can be found in `tests\Images\Input\Tga`

This is the output from imagesharp:
grayscale_a_UL_imagesharp

This is the output from imagemagick:
grayscale_a_UL_magick

All the failing tests are with 16 bit monochrome images.

### CODE REVIEW

**Comment by JimBobSquarePants** [comment]
@brianpopow Agree regarding output difference. Our output is what I would expect from 16bit grayscale.

File: src/ImageSharp/Formats/Tga/TgaDecoderCore.cs [hunk 1]
```diff
@@ -384,29 +413,65 @@ private void ReadMonoChrome<TPixel>(int width, int height, Buffer2D<TPixel> pixe
...
-                    Span<byte> rowSpan = row.GetSpan();
+                    int newY = InvertY(y, height, origin);
+                    Span<TPixel> pixelSpan = pixels.GetRowSpan(newY);
 
-                    if (!this.hasAlpha)
+                    if (isXInverted)
                     {
-                        // We need to set the alpha component value to fully opaque.
-                        for (int x = 1; x < rowSpan.Length; x += 2)
+                        for (int x = 0; x < width; x++)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Tga/TgaDecoderCore.cs]: We should move the conditionals outside the loop.

File: src/ImageSharp/Formats/Tga/TgaDecoderCore.cs [hunk 2]
```diff
@@ -504,7 +591,7 @@ private void ReadRle<TPixel>(int width, int height, Buffer2D<TPixel> pixels, int
                 this.UncompressRle(width, height, bufferSpan, bytesPerPixel);
                 for (int y = 0; y < height; y++)
                 {
-                    int newY = Invert(y, height, inverted);
+                    int newY = InvertY(y, height, origin);
                     Span<TPixel> pixelRow = pixels.GetRowSpan(newY);
                     int rowStartIdx = y * width * bytesPerPixel;
                     for (int x = 0; x < width; x++)
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Tga/TgaDecoderCore.cs]: These per-pixel conditionals will be hurting performance also.

**Comment by brianpopow** [comment]
> @brianpopow Agree regarding output difference. Our output is what I would expect from 16bit grayscale.

@JimBobSquarePants ok then i will add reference images to the ReferenceOutput folder and compare in the tests against those instead of the magick decoder output.

File: src/ImageSharp/Formats/Tga/TgaDecoderCore.cs (hunk 2 above)
Comment (brianpopow) [reply src/ImageSharp/Formats/Tga/TgaDecoderCore.cs]: i think i can move it outside the loop. Let me first get the tests green and then i will give it a try.

Comment (brianpopow) [reply src/ImageSharp/Formats/Tga/TgaDecoderCore.cs]: i have moved the calculation of the x and y positions out of the loop where possible, see: 70a62ead9e6bf66e2b4b85d2212be832380d8c7b.

The only place where i could not manage to do so (without making it overly complicated), was the RLE parts.

**Comment by brianpopow** [comment]
i have added more test cases to cover all the different variants.

File: src/ImageSharp/Formats/Tga/TgaDecoderCore.cs (hunk 2 above)
Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Tga/TgaDecoderCore.cs]: Yeah that's great 👍

**Pull request merged**